File: src/services/analyticsService.js
import api from './api';

export const analyticsService = {
  getAnalytics: async () => {
    const response = await api.get('/analytics');
    return response.data;
  },
  
  getSemesterAnalytics: async (semesterId) => {
    const response = await api.get(`/analytics/semester/${semesterId}`);
    return response.data;
  },
  
  getChartsData: async () => {
    const response = await api.get('/analytics/charts');
    return response.data;
  }
};

export default analyticsService;

File: student/src/main/java/com/studentdashboard/service/AnalyticsService.java
package com.studentdashboard.service;

import com.studentdashboard.dto.response.AnalyticsResponse;

public interface AnalyticsService {
    AnalyticsResponse getAnalytics(String username);
    AnalyticsResponse.SemesterData getSemesterAnalytics(String username, Long semesterId);
    Object getChartsData(String username);
}
